internal/collector/processor/queued_processor_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+)
+
+func TestQueueProcessorHappyPath(t *testing.T) {
+	wg := &sync.WaitGroup{}
+	mockProc := &mockConcurrentSpanProcessor{waitGroup: wg}
+	qp := NewQueuedSpanProcessor(mockProc)
+	goFn := func(b *agenttracepb.ExportTraceServiceRequest) {
+		qp.ProcessSpans(b, "test")
+	}
+
+	spans := []*tracepb.Span{{}}
+	wantBatches := 10
+	wg.Add(wantBatches)
+	wantSpans := 0
+	for i := 0; i < wantBatches; i++ {
+		batch := &agenttracepb.ExportTraceServiceRequest{
+			Spans: spans,
+		}
+		wantSpans += len(spans)
+		spans = append(spans, &tracepb.Span{})
+		go goFn(batch)
+	}
+
+	// Wait until all batches received
+	wg.Wait()
```

Comment:
I don't see any usage of `wg.Done` n times, looks like something is missing and I'd be surprised if we ever get past this point.

Reply:
Ditto